Add a FAQ to README, and comment the README's example (#154)
turion/rhine: README.md

  --   which is output every second.
  main :: IO ()
  main = flow $
    ms500 @@ waitClock            --  a Rhine = a ClSF in the context of a Clock

>     ||@ scheduleMillisecond @||   --  Compose 2 rhines in parallel

Reviewer: Consistent capitalisation

>     ||@ scheduleMillisecond @||   --  Compose 2 Rhines in parallel

Reviewer: ```suggestion
    ||@ scheduleMillisecond @||   --  compose 2 Rhines in parallel
```

* https://github.com/turion/rhine-tutorial: Presentation and tutorial app
* https://github.com/turion/sonnendemo: Demo application

## FAQ

> * Why does my blocking code, eg `arrMCl readLn`, behave [erratically](https://github.com/turion/rhine/issues/153)?

Reviewer: ```suggestion
* Why does my blocking code, e.g. `arrMCl readLn`, behave [erratically](https://github.com/turion/rhine/issues/153)?
```

Clocks must be the only thing that blocks the thread, not `ClSF`s. So for example, you can fix:

```haskell
arrMCl readLn
```

by using:

```haskell
tagS >>> arr read :: ClSF IO StdinClock () Int
```

> `tagS` contains the string that the `StdinClock` grabbed from `stdin`, and only the clock has been allowed to block the thread!

Reviewer: ```suggestion
[`tagS`](https://hackage.haskell.org/package/rhine/docs/FRP-Rhine.html#v:tagS) contains the string that the [`StdinClock`](https://hackage.haskell.org/package/rhine/docs/FRP-Rhine.html#t:StdinClock) grabbed from `stdin`, and only the clock has been allowed to block the thread!
```

I like to have links to reference docs, because few people will know all the Rhine components by heart.

Reviewer: These links could of course be applied everywhere in the documentation.

> * Can a sampling schedule dynamically change, eg depend on a signal?

Reviewer: ```suggestion
* Can a sampling schedule dynamically change, e.g. depend on a signal?
```

Yes, for instance you could implement a distance-dependent [collision detector](https://github.com/turion/rhine/issues/152).

> * How to handle slow computations, IE computations that take longer than the sample rate?

Reviewer: ```suggestion
* How to handle slow computations, i.e. computations that take longer than the sample rate?
```

> Several [strategies exist](https://github.com/turion/rhine/issues/151) and it depends on your use case. For `FixedStep` clocks, it won't matter since the execution of the program isn't tied to a realtime clock. For `ClSF`s running on `UTCTime` clocks, you can execute the slow code in a separate thread and coordinate merging the results back into the signal network.

Reviewer: ```suggestion
Several [strategies exist](https://github.com/turion/rhine/issues/151) and it depends on your use case.
For `FixedStep` clocks, it won't matter since the execution of the program isn't tied to a realtime clock.
For `ClSF`s running on `UTCTime` clocks, you can execute the slow code in a separate thread and coordinate merging the results back into the signal network.
```

Reviewer: (I prefer short, semantic lines. It makes diffs easier to read and merge.)
